From 87d354e8510072a1dcaa5698e4a0fbff34560ee5 Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Tue, 16 Nov 2010 13:41:45 +0000
Subject: Hide mounts having a path element starting with dot

Corresponds to commit 2bfcffde9ae5b14e78a4279bf48c71897c5f19d8 in glib.

Also bump required glib version to get consistent results.
---
diff --git a/monitor/gdu/ggduvolumemonitor.c b/monitor/gdu/ggduvolumemonitor.c
index 446acfc..8f0e643 100644
--- a/monitor/gdu/ggduvolumemonitor.c
+++ b/monitor/gdu/ggduvolumemonitor.c
@@ -638,6 +638,10 @@ _g_unix_mount_point_guess_should_display (GUnixMountPoint *mount_point)
   if (g_unix_is_mount_path_system_internal (mount_path))
     return FALSE;
 
+  /* Hide mounts within a dot path, suppose it was a purpose to hide this mount */
+  if (g_strstr_len (mount_path, -1, "/.") != NULL)
+    return FALSE;
+
   /* Only display things in /media (which are generally user mountable)
      and home dir (fuse stuff) */
   if (g_str_has_prefix (mount_path, "/media/"))
--
cgit v0.8.3.1
